docs/reference/mapping/runtime.asciidoc

```diff
@@ -747,6 +747,243 @@ address.
 // TESTRESPONSE[s/"_id" : "oWs5KXYB-XyJbifr9mrz"/"_id": $body.hits.hits.0._id/]
 // TESTRESPONSE[s/"day_of_week" : \[\n\s+"Sunday"\n\s\]/"day_of_week": $body.hits.hits.0.fields.day_of_week/]
 
+[[runtime-indexed]]
+=== Index a runtime field
+Runtime fields are defined by the context where they run. For example, you
+can define runtime fields in the
+<<runtime-search-request,context of a search query>> or within the
+<<runtime-mapping-fields,`runtime` section>> of an index mapping. If you
+decide to index a runtime field for greater performance, just move the full
+runtime field definition (including the script) to the context of an index
+mapping. This capability means you can write a script only once, and apply
+it to any context that supports runtime fields.
+
+NOTE: Currently, you can only index runtime fields of type `long` and
+`double`.
```

Comment:
meta note on the note :) given that we plan to add all the missing types rather quickly, is it worth to go and update each time this note, or shall we just remove it now?

Reply:
@javanna, I added the note because I wasn't sure if we're able to support all data types that support runtime fields for 7.13. If that's the plan, I'm happy to remove this note and reduce maintenance efforts.

Reply:
I think we may as well reflect exactly what it is in the branch now and then update it as we go. And remove it when we're done!

Reply:
I'd prefer removing the note, for simplicity, even if it does not reflect the current state of the branch.

Reply:
What happens if someone tries to index a runtime field of type `keyword`, or another data type that isn't yet supported? Or are we not concerned because this capability is in 7.x and not yet released? I just worry that a user tries to do this and contacts Support, and there's nothing in the docs saying "You can only index these types of runtime fields."

Reply:
we will support script for all types that are supported on the runtime section in 7.13.